spec/cordova/plugin/save.spec.js

```diff
@@ -54,15 +62,131 @@ describe('cordova/plugin/save', function () {
-        it('should return a version if a version was provided to plugin id');
-        it('should return a version that includes scope if scope was part of plugin id');
-        it('should fall back to using PluginInfoProvider to retrieve a version as last resort');
+        it('should return a plugin source\'s url or path property immediately', function () {
+            spyOn(save, 'getSpec').and.callThrough();
+            save.getSpec({ url: 'https://git-wip-us.apache.org/repos/asf/cordova-plugin-camera.git' }, '/some/path', 'VRPlugin');
+            expect(save.getSpec).toHaveBeenCalledWith(Object({ url: 'https://git-wip-us.apache.org/repos/asf/cordova-plugin-camera.git' }), '/some/path', 'VRPlugin');
+            expect(save.getSpec({ url: 'https://git-wip-us.apache.org/repos/asf/cordova-plugin-camera.git' }, '/some/path', 'VRPlugin')).toEqual('https://git-wip-us.apache.org/repos/asf/cordova-plugin-camera.git');
+        });
+
+        it('getSpec should return a version if a version was provided to plugin id', function () {
+            save.versionString.and.callThrough();
+            expect(save.getSpec({id: 'cordova-plugin-camera@^1.1.0'}, '/some/path', 'cordova-plugin-camera')).toEqual('^1.1.0');
+        });
+
+        it('should return a version that includes scope if scope was part of plugin id', function () {
+            save.versionString.and.callThrough();
+            expect(save.getSpec({ type: 'registry', id: '@scoped/package@^1.0.0' }, '/some/path', 'cordova-plugin-camera')).toEqual('@scoped/package@^1.0.0');
+        });
+
+        it('should fall back to using PluginInfoProvider to retrieve a version as last resort', function () {
+            expect(save.getSpec({ id: 'cordova-plugin-camera' }, '/some/path', 'cordova-plugin-camera')).toEqual(null);
+            expect(plugin_info_provider_mock.prototype.get).toHaveBeenCalled();
+        });
+    });
+
+    describe('getPluginVariables helper method', function () {
+        it('if no variables are passed in, should return empty', function () {
+            expect(save.getPluginVariables()).toEqual([]);
+        });
+        it('if variables are passed in, should return result & get added to name and value', function () {
+            expect(save.getPluginVariables({ variable: 'var 1' })).toEqual([ { name: 'variable', value: 'var 1' } ]);
+        });
+    });
+
+    describe('versionString helper method', function () {
+        it('if no version, should return null', function () {
+            save.versionString.and.callThrough();
+            spyOn(semver, 'valid').and.returnValue('', false);
+            spyOn(semver, 'validRange').and.returnValue('', false);
+            expect(save.versionString()).toBe(null);
+        });
+        it('return version passed in, if it is within the valid range', function () {
+            save.versionString.and.callThrough();
+            spyOn(semver, 'valid').and.returnValue('', false);
```

Comment:
`semver.valid` returns null when the version isn't valid (just looked it up at https://github.com/npm/node-semver.
```
spyOn(semver, 'valid').and.returnValue(null);
```